File: C++/maximum-twin-sum-of-a-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    int pairSum(ListNode* head) {
        vector<int> data;
        
        for (ListNode* node = head; node; node = node->next) {
            data.push_back(node->val);
        }
        
        int maxSum = INT32_MIN;
        int sz = data.size();
        
        for (int i = 0, j = sz - 1; i < j; i++, j--) {
            int sum = data[i] + data[j];
            if (maxSum < sum) {
                maxSum = sum;
            }
        }
        
        return maxSum;
    }
};

File: JavaScript/maximum-twin-sum-of-a-linked-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {number}
 */
var pairSum = function(head) {
    const values = []
    for (let node = head; node; node = node.next) {
        values.push(node.val)
    }
    let maxSum = values[0] + values[values.length - 1]
    for (let i = 0, j = values.length - 1; i < j; i++, j--) {
        const sum = values[i] + values[j]
        if (maxSum < sum)
            maxSum = sum
    }
    return maxSum
};